packages/react-native-branch/src/withBranchAndroid.ts

```diff
@@ -0,0 +1,339 @@
+  ];
+
+  return mergeContents({
+    tag: "react-native-branch-import",
+    src,
+    newSrc: newSrc.join("\n"),
+    anchor: `package ${packageId};`,
+    offset: 1,
+    comment: "//",
+  });
+}
+
+export function addBranchInitSession(src: string): MergeResults {
+  const tag = "react-native-branch-init-session";
+
+  try {
+    const newSrc = [
+      "    RNBranchModule.initSession(getIntent().getData(), this);",
+    ];
+
+    return mergeContents({
+      tag,
+      src,
+      newSrc: newSrc.join("\n"),
+      anchor: /super\.onStart\(\);/,
+      offset: 1,
+      comment: "//",
+    });
+  } catch (err) {
+    if ((err as any).code !== "ERR_NO_MATCH") {
+      throw err;
+    }
+  }
+
+  const newSrc = [
+    "  @Override",
+    "  protected void onStart() {",
+    "    super.onStart();",
+    "    RNBranchModule.initSession(getIntent().getData(), this);",
+    "  }",
+  ];
+  return mergeContents({
+    tag,
+    src,
+    newSrc: newSrc.join("\n"),
+    anchor: `getMainComponentName`,
+    offset: 3,
+    comment: "//",
+  });
+}
+
+export function addBranchOnNewIntent(src: string): MergeResults {
+  const tag = "react-native-branch-on-new-intent";
+
+  try {
+    const newSrc = ["    RNBranchModule.onNewIntent(intent);"];
+
+    return mergeContents({
+      tag,
+      src,
+      newSrc: newSrc.join("\n"),
+      anchor: /super\.onNewIntent\(intent\);/,
+      offset: 1,
+      comment: "//",
+    });
+  } catch (err) {
+    if ((err as any).code !== "ERR_NO_MATCH") {
+      throw err;
+    }
+  }
+
+  const newSrc = [
+    "  @Override",
+    "  public void onNewIntent(Intent intent) {",
+    "    super.onNewIntent(intent);",
+    "    RNBranchModule.onNewIntent(intent);",
+    "  }",
+  ];
+  return mergeContents({
+    tag,
+    src,
+    newSrc: newSrc.join("\n"),
+    anchor: `getMainComponentName`,
+    offset: 3,
+    comment: "//",
+  });
+}
+
+export function getBranchApiKey(config: ExpoConfig) {
+  return config.android?.config?.branch?.apiKey ?? null;
+}
+
+export function setBranchApiKey(
+  apiKey: string,
+  androidManifest: AndroidConfig.Manifest.AndroidManifest
+) {
+  const mainApplication = getMainApplicationOrThrow(androidManifest);
+
+  if (apiKey) {
+    // If the item exists, add it back
+    addMetaDataItemToMainApplication(mainApplication, META_BRANCH_KEY, apiKey);
+  } else {
+    // Remove any existing item
+    removeMetaDataItemFromMainApplication(mainApplication, META_BRANCH_KEY);
+  }
+
+  return androidManifest;
+}
+
+export const withBranchAndroid: ConfigPlugin<{ apiKey?: string }> = (
+  config,
+  data
+) => {
+  const apiKey = data.apiKey ?? getBranchApiKey(config);
+  if (!apiKey) {
+    throw new Error("Branch API key is required");
+  }
+
+  config = withAndroidManifest(config, (config) => {
+    config.modResults = setBranchApiKey(apiKey, config.modResults);
+    return config;
+  });
+
+  // Directly edit MainApplication.java
+  config = withDangerousMod(config, [
```

Comment:
We support `withMainApplication` due to the package name resolution https://docs.expo.dev/guides/config-plugins/#mod-plugins

Reply:
Hey @EvanBacon, which APIs I should use? `mergeContents` or this?
```
import {
  addImports,
  appendContentsInsideDeclarationBlock,
  findNewInstanceCodeBlock,
} from "@expo/config-plugins/build/android/codeMod";
```